zhaquirks/sonoff/snzb02d.py

```diff
@@ -0,0 +1,147 @@
+"""Module for Philips quirks implementations."""
+from __future__ import annotations
+
+from typing import Any, Final
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.types.basic import enum_factory
+from zigpy.zcl.clusters.general import (
+    Basic,
+    Identify,
+    Ota,
+    PollControl,
+    PowerConfiguration,
+)
+from zigpy.zcl.clusters.measurement import RelativeHumidity, TemperatureMeasurement
+from zigpy.zcl.foundation import (
+    BaseAttributeDefs,
+    Direction,
+    GeneralCommand,
+    ZCLAttributeDef,
+    ZCLHeader,
+)
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+WWAH_CLUSTER_ID = 0xFC57
+
+
+class DisplayUnit(enum_factory(t.uint16_t)):
+    Celsius = 0
+    Fahrenheit = 1
+
+
+class SonoffClusterFC11(CustomCluster):
+    """Sonoff custom cluster."""
+
+    DisplayUnit: Final = DisplayUnit
+
+    cluster_id = 0xFC11
+    name = "SonoffClusterFC11"
+
+    class AttributeDefs(BaseAttributeDefs):
+        high_temperature_threshold: Final = ZCLAttributeDef(
+            id=0x0003, type=t.int16s, access="rwp", is_manufacturer_specific=True
+        )
+        low_temperature_threshold: Final = ZCLAttributeDef(
+            id=0x0004, type=t.int16s, access="rwp", is_manufacturer_specific=True
+        )
+        low_humidity_threshold: Final = ZCLAttributeDef(
+            id=0x0005, type=t.uint16_t, access="rwp", is_manufacturer_specific=True
+        )
+        high_humidity_threshold: Final = ZCLAttributeDef(
+            id=0x0006, type=t.uint16_t, access="rwp", is_manufacturer_specific=True
+        )
+        display_unit: Final = ZCLAttributeDef(
+            id=0x0007, type=DisplayUnit, access="rwp", is_manufacturer_specific=True
+        )
+        unknown_fffd: Final = ZCLAttributeDef(
+            id=0xFFFD, type=t.uint16_t, access="rwp", is_manufacturer_specific=True
+        )
+
+    def _create_request(
+        self,
+        *,
+        general: bool,
+        command_id: GeneralCommand | int,
+        schema: dict | t.Struct,
+        manufacturer: int | None = None,
+        tsn: int | None = None,
+        disable_default_response: bool,
+        direction: Direction,
+        # Schema args and kwargs
+        args: tuple[Any, ...],
+        kwargs: Any,
+    ) -> tuple[ZCLHeader, bytes]:
+        """Override all request to disable manufacturer."""
```

Comment:
I tried higher level overrides, like skipping the CustomCluster. But seems HA overrides the possibility of using None as signifier of letting the customcluster decide on use here: https://github.com/home-assistant/core/blob/10014838ef32ee1607e3b959c43821a887c5e3ea/homeassistant/components/zha/websocket_api.py#L823

Reply:
The linked code only supports the manage device dialog IIRC and it’s a sensible default. 99% of the time you need it for mfg specific things and common users aren’t going to be aware of that. 

I think what you are doing here is the only way to achieve this currently.

Reply:
The CustomCluster has this default too. Would make more sense for backend to handle it. Zigpy or Quirks.

Also as it stands now its impossible to call with no manufacturer code using that dialog.